### PR TITLE
feat: complete current user display

### PR DESCRIPTION
**Pull request by GabrielOverflow**

### CODE REVIEW

File: pages/profile.js [hunk 1]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
 import ProfileFollow from '../components/Profile/ProfileFollow';
 import UserAdd from '../icons/user-add';
-import { follow } from '../services/followServices';
+import { follow, getFollowedStatus } from '../services/followServices';
+import { getCurrentUser } from '../services/usersServices';
 import hotToast from '../utils/hotToast';
 import newNotification from '../utils/newNotification';
 
-const Profile = () => {
+const Profile = (props) => {
```
Comment (devharris7) [inline pages/profile.js]: = ({ username })

Comment (devharris7) [reply pages/profile.js]: add propTypes

File: pages/profile.js [hunk 2]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+  const { username } = props;
   const [currentTab, setCurrentTab] = useState('posts');
   const [followedStatus, setFollowedStatus] = useState('not_followed');
+  const [currentName, setCurrentName] = useState('null');
```
Comment (devharris7) [inline pages/profile.js]: useState('')

File: pages/profile.js [hunk 3]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+  // const [role, setRole] = useState('null');
+  const [currentRole, setCurrentRole] = useState('null');
```
Comment (devharris7) [inline pages/profile.js]: useState('')

File: pages/profile.js [hunk 4]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+
+  // Get current user details
+  useEffect(() => {
+    const getUser = async () => {
+      const response = await getCurrentUser();
```
Comment (devharris7) [inline pages/profile.js]: const { data }

File: pages/profile.js [hunk 5]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+      console.log(response.data);
+      setCurrentName(response.data.username);
+      setCurrentRole(response.data.role.roleName);
+      if (username === undefined) {
```
Comment (devharris7) [inline pages/profile.js]: if(!username)

File: pages/profile.js [hunk 6]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+        setFollowedStatus('current_user');
+      }
+    };
+    getUser();
+  }, [username]);
+
+  // Check follow status
+  useEffect(() => {
+    const checkStatus = async (name) => {
+      const response = await getFollowedStatus(name);
```
Comment (devharris7) [inline pages/profile.js]: const {data}

File: pages/profile.js [hunk 7]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+      if (response.data === true) {
```
Comment (devharris7) [inline pages/profile.js]: if (data)

File: pages/profile.js [hunk 8]
```diff
@@ -14,13 +14,46 @@ import { blueGrey } from '@mui/material/colors';
...
+        console.log('you have followed this user');
+        setFollowedStatus('followed');
+      }
+    };
+    if (username !== undefined) {
```
Comment (devharris7) [inline pages/profile.js]: if (username)

File: pages/category/[categoryTitle].js [hunk 9]
```diff
@@ -341,9 +328,9 @@ const PostList = ({ categoryInfo, initialTotalCount, pinPostInfo }) => {
             value={sortColumn}
             onChange={handleSortColumn}
           >
-            {postSortColumns.map((option) => (
-              <MenuItem key={option.value} value={option.value}>
-                {option.value}
+            {Object.keys(sortColumnList).map((option) => (
```
Comment (devharris7) [inline pages/category/[categoryTitle].js]:
```
const menuItems = React.useMemo(()=>Object.keys(sortColumnList).map((option) => ( <MenuItem key={option} value={option}>
                {option}
              </MenuItem>
            ), [sortColumnList])
```

{menuItems}

File: pages/profile.js [hunk 10]
```diff
@@ -14,13 +14,45 @@ import { blueGrey } from '@mui/material/colors';
...
+      }
+    };
+    getUser();
+  }, [username]);
+
+  // Check follow status
+  useEffect(() => {
+    const checkStatus = async (name) => {
+      const { data } = await getFollowedStatus(name);
+      if (data === true) {
```
Comment (devharris7) [inline pages/profile.js:46]: what is the data type? must this data be `true`, or you mean the `data` exists or not

File: pages/profile.js [hunk 11]
```diff
@@ -124,10 +156,23 @@ const Profile = () => {
             }}
           />
           <Box sx={{ ml: 2 }}>
-            <Typography color="textSecondary" variant="overline">
-              {profile.title}
-            </Typography>
-            <Typography variant="h6">{profile.name}</Typography>
+            {username === undefined && (
```
Comment (devharris7) [inline pages/profile.js]: dont use `username === undefined`

use `!username`

File: pages/profile.js [hunk 12]
```diff
@@ -124,10 +156,23 @@ const Profile = () => {
...
+                {currentRole}
+              </Typography>
+            )}
+            {/* 这里还有待修改 */}
+            {username !== undefined && (
+              <Typography color="textSecondary" variant="overline">
+                {profile.title}
+              </Typography>
+            )}
+            {username === undefined && (
```
Comment (devharris7) [inline pages/profile.js]: refactor l170 - 175 to 

```
<Typography variant="h6">{username ? username : currentName}</Typography>
```

File: services/Users.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+export const uniqueEmail = (email) =>
+  http(`/v1/users/unique-email/${email}`, { method: 'get' });
+
+export const uniqueUsername = (username) =>
+  http(`/v1/users/unique-username/${username}`, { method: 'get' });
+
+export const sendResetPasswordEmail = (email) =>
+  http(`/v1/users/reset-password/${email}`, { method: 'GET' });
+
+export const resetPassword = (password) =>
```
Comment (devharris7) [inline services/Users.js]: you can't pass password as a url param, consider security

File: services/Users.js [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import http from '../utils/axios';
+
+export const login = (email, password) =>
+  http(`/login`, { method: 'POST', data: { email, password } });
+
+export const register = (email, username, password) =>
+  http(`/v1/users/register/${email}/${username}/${password}`, {
```
Comment (devharris7) [inline services/Users.js]: you can't pass password as a url param, consider security

File: pages/profile.js (hunk 10 above)
Comment (GabrielOverflow) [reply pages/profile.js:46]: The data type is Boolean. Should it be "if (data) {"?

**Pull request merged**